app/web/src/helpers/hooks/useIdeState.js

```diff
@@ -138,17 +145,22 @@ export const requestRender = ({
   code,
   camera,
   viewerSize,
+  specialCadProcess = null,
 }) => {
-  state.ideType !== 'INIT' &&
-    !state.isLoading &&
-    cadPackages[state.ideType]
-      .render({
-        code,
-        settings: {
-          camera,
-          viewerSize,
-        },
-      })
+  if (
+    state.ideType !== 'INIT' &&
+    (!state.isLoading || state.objectData?.type === 'INIT')
+  ) {
+    const renderFn = specialCadProcess
+      ? cadPackages[state.ideType][specialCadProcess]
+      : cadPackages[state.ideType].render
```

Comment:
pretty messy. this is so that running the stl function rather than preview for openscad when the user want's to download the stl.